### PR TITLE
fix(controller): Only warn if cron job missing. Fixes #4351

### LINKED ISSUE #4351: v2.11.6: Parent CronWorkflow 'argo/canary' is malformed

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [x] I've signed the CLA.
* [ ] ~~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~~
* [x] My builds are green. Try syncing with master if they are not. 

Fixes #4351

### CODE REVIEW

**Review by sarabala1979 — APPROVED** [review]

**Comment by alexec** [comment]
@sarabala1979 can I ask you to review again please? I've re-implemented this to design out the problem.

File: workflow/cron/cron-facade.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	defer f.mu.Unlock()
+	entryID, ok := f.entryIDs[key]
+	if !ok {
+		return
+	}
+	f.cron.Remove(entryID)
+	delete(f.entryIDs, key)
+}
+
+func (f *cronFacade) AddJob(key, spec string, ctx *cronWfOperationCtx) error {
```
Comment (simster7) [inline workflow/cron/cron-facade.go:45]:
```suggestion
func (f *cronFacade) AddJob(key, schedule string, ctx *cronWfOperationCtx) error {
```
perhaps more clear?

File: workflow/cron/cron-facade.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package cron
```
Comment (simster7) [inline workflow/cron/cron-facade.go:1]: Use `_` in file name to keep with Go standard
```diff
- workflow/cron/cron-facade.go
+ workflow/cron/cron_facade.go
```

**Review by simster7 — APPROVED** [review]
LGTM! Two tiny comments

File: workflow/cron/cron-facade.go (hunk 2 above)
Comment (alexec) [reply workflow/cron/cron-facade.go]: done

**Pull request merged**